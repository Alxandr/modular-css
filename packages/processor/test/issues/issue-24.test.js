"use strict";

var dedent = require("dedent"),
    namer  = require("test-utils/namer.js"),

    Processor = require("../../processor.js");

describe("/issues", () => {
    describe("/24", () => {
        it("should be able to compose using a value", () => {
            var processor = new Processor({
                    namer,
                });
            
            return processor.string(
                "./packages/core/test/specimens/composition.css",
                dedent(`
                    @value simple: "./simple.css";
                    
                    .a {
                        composes: wooga from simple;
                        background: #000;
                    }
                `)
            )
            .then((result) => {
                expect(result.exports).toMatchSnapshot();

                return processor.output();
            })
            .then((result) => expect(result.css).toMatchSnapshot());
        });
    });
});
