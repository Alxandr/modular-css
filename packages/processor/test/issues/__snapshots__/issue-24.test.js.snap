// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/issues /24 should be able to compose using a value 1`] = `
Object {
  "a": Array [
    "wooga",
    "a",
  ],
  "simple": "\\"./simple.css\\"",
}
`;

exports[`/issues /24 should be able to compose using a value 2`] = `
"/* packages/processor/test/specimens/simple.css */
.wooga { color: red; }
/* packages/processor/test/specimens/composition.css */
.a {
    background: #000;
}
"
`;
